app/server/src/controllers/appsController.ts

```diff
@@ -2,14 +2,23 @@ import { Request, Response } from "express";
 import { AppLocals } from "../types";
 import { ErrorType } from "../errors/errorType";
 import { WodinWebError } from "../errors/wodinWebError";
+import { getSessionStore } from "../db/sessionStore";
 
 export class AppsController {
-    static getApp = (req: Request, res: Response) => {
+    static getApp = async (req: Request, res: Response) => {
         const {
             configReader, appsPath, wodinConfig, wodinVersion
         } = req.app.locals as AppLocals;
         const { appName } = req.params;
-        const { sessionId } = req.query;
+        // client can pass either sessionId or share (friendly id) parameter to identify session to reload
+        let sessionId = req.query.sessionId as string | undefined | null;
+        const { share } = req.query;
+
+        if (share) {
+            const store = getSessionStore(req);
+            sessionId = await store.getSessionIdFromFriendlyId(share as string);
```

Comment:
if this fails, we should not just load the default without some indication; so going to `http://localhost:3000/apps/day2/?share=criminal-antelope` should show an error like "share code 'criminal-antelope'" not found

More generally and possibly less usefully going to to `?a=b` might also indicate that the query was unexpected. This might happen in the case where people are trying to remember/guess how the urls work

Reply:
Good idea. It seemed like the easiest way to do this would be to communicate a not found share id into the top level component through a prop, and then let it use the generic errors module to add an error which is shown in our standard dismissble error alert. 

Not sure about the more general question. I guess we could show some sort of info message if we get a query parameter we weren't expecting and listing the ones we are?